src/main/java/com/linagora/james/mailets/json/MailData.java

```diff
@@ -0,0 +1,107 @@
+ * This program is free software: you can redistribute it and/or modify        *
+ * it under the terms of the GNU Affero General Public License as published by *
+ * the Free Software Foundation, either version 3 of the License, or           *
+ * (at your option) any later version.                                         *
+ *                                                                             *
+ * This program is distributed in the hope that it will be useful,             *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the               *
+ * GNU Affero General Public License for more details.                         *
+ *                                                                             *
+ * You should have received a copy of the GNU Affero General Public License    *
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.       *
+ *******************************************************************************/
+package com.linagora.james.mailets.json;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.james.mime4j.dom.Message;
+import org.apache.james.mime4j.message.MessageBuilder;
+import org.apache.james.util.mime.MessageContentExtractor;
+import org.apache.james.util.mime.MessageContentExtractor.MessageContent;
+import org.apache.mailet.Mail;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+
+public class MailData {
+
+    public static MailData from(Mail mail, UUID messageId) throws MessagingException, IOException {
+        MimeMessage message = mail.getMessage();
+        
+        return new MailData(messageId,
+                Emailers.from(message.getFrom()),
+                Recipients.from(message),
+                ImmutableList.of(Optional.ofNullable(message.getSubject()).orElse("")),
+                retrieveTextPart(mail));
+    }
+
+    private static String retrieveTextPart(Mail mail) throws IOException, MessagingException {
+        MessageContent messageContent = new MessageContentExtractor()
+                .extract(toMime4jMessage(mail));
+        return mainTextContent(messageContent).orElse("");
+    }
+
+    private static Message toMime4jMessage(Mail mail) throws IOException, MessagingException, UnsupportedEncodingException {
+        ByteArrayOutputStream rawMessage = new ByteArrayOutputStream();
+        mail.getMessage().writeTo(rawMessage);
+        return MessageBuilder
+                .read(IOUtils.toInputStream(rawMessage.toString(Charsets.UTF_8.name())))
```

Comment:
this is very inefficient. You create a byte array with the whole mail, then a string with the same thing encoded in UTF8. I'm sure you can create a stream for the byte array